### PR TITLE
[serve][docs] Update key concepts page for Ray 2.0

### LINKED ISSUE #27438: [serve][docs] Update "Key Concepts" page
Deployment
- Good.

HTTP Ingress
- Remove. We should replace this with “Ingress Deployment” instead (which is the top-level deployment of the application).
- Drop `route_prefix` from here. Instead let’s have a tutorial on routing to downstreams.

ServeHandle
- Remove the extremely long `NOTE` and put it into a user guide instead.

DeploymentGraph
- Good.

### PR DESCRIPTION
**Pull request by edoakes**
Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>

## Why are these changes needed?

Closes https://github.com/ray-project/ray/issues/27438

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by sihanwang41 — APPROVED** [review]

File: doc/source/serve/key-concepts.md [hunk 1]
```diff
@@ -24,91 +25,100 @@ class MyFirstDeployment:
...
-By default, HTTP requests will be forwarded to the `__call__` method of the class (or the function) and a `Starlette Request` object will be the sole argument.
-You can also define a deployment that wraps a FastAPI app for more flexible handling of HTTP requests. See {ref}`serve-fastapi-http` for details.
+(serve-key-concepts-query-deployment)=
+
+## ServeHandle (composing deployments)
 
-To deploy multiple deployments that serve the same class or function, use the `name` option:
+Ray Serve enables flexible model composition and scaling by allowing multiple independent deployments to call into each other.
+When binding a deployment, included in the arguments can be references to _other bound deployments_.
+Then, at runtime this argument is converted to a {mod}`ServeHandle <ray.serve.handle.RayServeHandle>` that can be used to query the deployment using a Python-native API.
```
Comment (dmatrix) [inline doc/source/serve/key-concepts.md:39]: When binding a deployment, included in the arguments, can be ....

**Review by dmatrix — APPROVED** [review]
There are couple places where you can use present tense instead of future but it's fine. LGTM

File: doc/source/serve/key-concepts.md (hunk 1 above)
Comment (edoakes) [reply doc/source/serve/key-concepts.md:39]: thanks, will fix!

**Comment by edoakes** [comment]
@simon-mo PTAL

File: doc/source/serve/key-concepts.md [hunk 2]
```diff
@@ -24,91 +25,100 @@ class MyFirstDeployment:
...
+        
+
+model_a = ModelA.bind()
+model_b = ModelB.bind()
+
+# model_a and model_b will be passed to the Driver constructor as ServeHandles.
+driver = Driver.bind(model_a, model_b)
+
+# Deploys model_a, model_b, and driver.
+serve.run(driver)
```
Comment (simon-mo) [inline doc/source/serve/key-concepts.md:62]: show the calling example? as well as moving these to doc_code?

Comment (simon-mo) [reply doc/source/serve/key-concepts.md:62]: do you want to dive into the two types of handles? or that belong to a separate page?

Comment (edoakes) [reply doc/source/serve/key-concepts.md:62]: I think that should belong on a separate page(s). The dev workflow should talk about calling driver from python (cc @architkulkarni) and the model composition page should talk about calling deployments from one another (cc @shrekris-anyscale).

**Pull request merged**